### ISSUE TITLE
Using hdf5 to store pdb coordinate of multiple PDB with different length, single run will analyze them all

### ISSUE THREAD

**Issue by vuqv** [qvv5013/entanglement_analysis#2]
This is needed since Julia is fast but the time to initialize env for running program takes ~5-10s. If we want to analyze 10,000 PDB structures by running 10,000 times, it is 50,000 s for doing nonsense thing.

This is equivalent to analyze DCD file but in DCD, all frames have the same shape of coordinate matrix, it initializes env only once.

To read coordinates from 1000 PDB files and save their coordinates to an HDF5 file in Julia, you can use the BioStructures package to read the PDB files and the HDF5 package to save the coordinates.

#### Explanation:
* `read_coordinates(pdb_file)`: This function reads a single PDB file and extracts the atomic coordinates.
* `read_all_coordinates(pdb_files)`: This function reads multiple PDB files and collects all their coordinates into a list.
* `save_coordinates_to_hdf5(hdf5_file, all_coords)`: This function saves the collected coordinates to an HDF5 file, where each dataset in the HDF5 file corresponds to the coordinates from one PDB file.
Make sure to adjust the list pdb_files to include the paths to all 1000 of your PDB files. The above script will read the coordinates from each PDB file and store them efficiently in an HDF5 file.

`Notes: Current version of Julia I am using is 1.8.5 which will be compatible with BioStructure 2.1.0`

```julia
import BioStructures as bs
using HDF5
function read_coordinates(pdb_file)
    # load PDB file
    structure = bs.read(pdb_file, bs.PDB)
    # Number of CA atoms
    n_residues = bs.countatoms(structure, bs.calphaselector)
    # get coordinate of all CA atoms
    coords = [atom.coords for atom in bs.collectatoms(structure, bs.calphaselector)]
    # Processing coordinate
    # Flatten
    flatten_coords = vcat(coords...)
    # reshape to 3xN_atoms
    mat_coords = reshape(flatten_coords, 3, n_residues)
    return mat_coords
end
function read_coordiantes_multiple_files(pdb_files)
    all_coords = []
    for pdb_file in pdb_files
        pdb_file_coordinates = read_coordinates(pdb_file)
        push!(all_coords, pdb_file_coordinates)
    end
    return all_coords
end
function save_coordinates_to_hdf5(hdf5_file, all_coords)
    h5write(hdf5_file) do file
        for (i, coords) in enumerate(all_coords)
            dataset_name = "coordinates$i"
            write(file, dataset_name, coords)
        end
    end
end
# List of PDB files (assuming the files are in the current directory or provide the full path)
pdb_files = ["file1.pdb", "file2.pdb", "file3.pdb"]  # Add all 1000 file names here
# Read coordinates from all PDB files
all_coords = read_all_coordinates(pdb_files)
# Save coordinates to an HDF5 file
save_coordinates_to_hdf5("coordinates.h5", all_coords)
```

### After saving coordiantes to hdf5, load hdf5 (separated task) and doing calculation for every element of hdf5:

To load an HDF5 file containing the coordinates of multiple PDB files and perform calculations on those coordinates, you can follow a structured approach. Here’s a detailed example:
#### Step-by-Step Process:
* Read the HDF5 file and extract the coordinates for each PDB entry.
* Perform calculations on the extracted coordinates.
* Wrap the process in functions for clarity and reusability.

#### Explanation:
* Loading Coordinates from HDF5:
load_coordinates_from_hdf5 function opens the HDF5 file, reads all datasets, and stores the coordinates in a list.
* Performing Calculations:
calculate_center_of_mass function calculates the center of mass for a set of coordinates.
process_all_coordinates function processes all the coordinates loaded from the HDF5 file, applying the calculate_center_of_mass function to each set and storing the results.

#### Main Script:
* Loads the coordinates from the HDF5 file.
* Calculates the center of mass for each set of coordinates.
* Prints the results.

```julia
using HDF5
using LinearAlgebra
# Function to load coordinates from HDF5 file
function load_coordinates_from_hdf5(hdf5_file::String)
    all_coords = []
    h5open(hdf5_file, "r") do file
        dataset_names = keys(file)  # Get all dataset names
        for name in dataset_names
            push!(all_coords, read(file, name))
        end
    end
    return all_coords
end
# Example function to calculate the center of mass for each set of coordinates
function calculate_center_of_mass(coords::Vector{Tuple{Float64, Float64, Float64}})
    n = length(coords)
    center_of_mass = sum(coords) / n
    return center_of_mass
end
# Function to process all coordinates
function process_all_coordinates(hdf5_file::String)
    all_coords = load_coordinates_from_hdf5(hdf5_file)
    results = []
    for coords in all_coords
        center_of_mass = calculate_center_of_mass(coords)
        push!(results, center_of_mass)
    end
    return results
end
# Example usage
hdf5_file = "coordinates.h5"
results = process_all_coordinates(hdf5_file)
# Print results (centers of mass for each set of coordinates)
for (i, result) in enumerate(results)
    println("Center of Mass for PDB $i: $result")
end
```

For formalize and easy to read code, using `import` alias instead of flushing all module functions to env:

Using the import statement with an alias:
import allows you to bring in the module itself, requiring you to qualify symbols with the module name. You can give the module an alias for convenience.

 ```julia
module MyModule
    export some_function
    some_function() = println("Hello from MyModule!")
end
import .MyModule as mdl
mdl.some_function()  # Using the alias to call the function
```

In the first case, using using .MyModule makes the some_function directly available without qualification. In the second case, using import .MyModule as mdl allows you to call the function via the alias mdl.
To summarize, to call a function from a module with an alias in Julia, you can use:
```julia
import .ModuleName as alias
alias.function_name()
```
This approach is quite similar to how you would do it in Python with import module_name as mdl.
